content/spin/kv-store.md

```diff
@@ -0,0 +1,355 @@
+					http.StatusInternalServerError,
+				)
+				return
+			}
+		}
+
+		if err := key_value.Delete(store, "foo"); err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		if exists, err := key_value.Exists(store, "foo"); err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		} else if exists {
+			http.Error(w, "key was not deleted as expected", http.StatusInternalServerError)
+			return
+		}
+	})
+}
+
+func main() {}
+
+```
+
+{{ blockEnd }}
+
+{{ blockEnd }}
+
+## Configure `spin.toml` To Use A Default Key/Value Store
+
+{{ tabs "sdk-type" }}
+
+{{ startTab "Rust"}}
+
+<!-- @selectiveCpy -->
+
+```toml
+
+spin_version = "1"
+authors = ["Fermyon Engineering <engineering@fermyon.com>"]
+description = "A simple application that exercises key-value storage."
+name = "spin-key-value"
+trigger = {type = "http", base = "/test"}
+version = "1.0.0"
+
+[[component]]
+id = "hello"
+source = "target/wasm32-wasi/release/spin_key_value.wasm"
+key_value_stores = ["default"]
+[component.trigger]
+route = "/..."
+[component.build]
+command = "cargo build --target wasm32-wasi --release"
+
+```
+
+{{ blockEnd }}
+
+{{ startTab "TypeScript" }}
+
+```toml
+
+spin_version = "1"
+authors = ["Fermyon Engineering <engineering@fermyon.com>"]
+description = "A simple application that exercises key-value storage."
+name = "spin-key-value"
+trigger = {type = "http", base = "/test"}
+version = "1.0.0"
+
+[[component]]
+id = "hello"
+source = "target/wasm32-wasi/release/spin_key_value.wasm"
+key_value_stores = ["default"]
+[component.trigger]
+route = "/..."
+[component.build]
+command = "npm run build"
+
+```
+
+{{ blockEnd }}
+
+{{ startTab "TinyGo" }}
+
+```toml
+
+spin_version = "1"
+authors = ["Fermyon Engineering <engineering@fermyon.com>"]
+name = "tinygo-key-value-example"
+trigger = { type = "http", base = "/" }
+version = "0.1.0"
+
+[[component]]
+id = "key-value"
+source = "main.wasm"
+key_value_stores = ["default"]
+[component.trigger]
+route = "/test"
+[component.build]
+command = "tinygo build -target=wasi -gc=leaking -no-debug -o main.wasm main.go"
+
+```
+
+{{ blockEnd }}
+
+{{ blockEnd }}
+
+## Build And Deploy Your Spin Application
+
+Now let's build and deploy our Spin Application locally. Run the following command to build your application: 
+
+<!-- @selectiveCpy -->
+
+```bash
+
+spin build
+
+```
+
+Now run the subsequent command to deploy your application: 
+
+<!-- @selectiveCpy -->
+
+```bash
+
+spin up
+
+```
+
+## Store and Retrieve Data From Your Default Key/Value Store
+
+Once you have completed this minimal configuration and deployed your application, data will be persisted across requests:
+
+<!-- @selectiveCpy -->
+
+```bash
+
+curl localhost:3000/hello
+curl localhost:3000/goodbye
+curl localhost:3000/hi-again
+
+```
+
+You should see the following output to confirm the key/value pairs are being stored correctly:
+
+<!-- @nocopy -->
```

Comment:
Are we sure that we do not want people copying these commands? I do see the same commands listed above, maybe we can combine both so as to not be repetitive.

Reply:
That's a great point @karthik2804 I will update this in a new push directly to @macolso's `kvstore` branch and then resolve this conversation here in the GH UI.